exporters/deploytime/collector_github.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+import logging
+import re
+from datetime import datetime
+from typing import Any, Iterable, NamedTuple, Optional, cast
+
+from prometheus_client.core import GaugeMetricFamily
+from prometheus_client.registry import Collector
+from requests import Session
+
+import pelorus
+from pelorus.utils import TokenAuth, join_url_path_components
+from provider_common.github import GitHubError, paginate_github, parse_datetime
+
+
+class Release(NamedTuple):
+    tag_name: str
+    published_at: datetime
+
+    @classmethod
+    def from_json(cls, json_object: dict[str, Any]) -> Release:
+        tag_name = json_object["tag_name"]
+        published_at = parse_datetime(json_object["published_at"])
+
+        return Release(tag_name, published_at)
+
+
+class GitHubReleaseCollector(Collector):
+    # TODO: regex to determine which releases are "prod"?
+
+    def __init__(
+        self,
+        projects: Iterable[str],
+        custom_host: str = "",
+        apikey: Optional[str] = None,
```

Comment:
AFAICT the env var TOKEN is not being read in here. Line 43 ( if apikey) is None.